projects/keep3r.js

```diff
@@ -1,34 +1,183 @@
-    return { [keep3rAddress] : bondedKeep3r };
+
+// Iron Bank Collateral
+const cTokens = {
+  CYEUR: "0x00e5c0774a5f065c285068170b20393925c84bf3",
+  CYJPY: "0x215F34af6557A6598DbdA9aa11cc556F5AE264B1",
+  CYGBP: "0xecaB2C76f1A8359A06fAB5fA0CEea51280A97eCF",
+  CYCHF: "0x1b3E95E8ECF7A7caB6c4De1b344F94865aBD12d5",
+  CYAUD: "0x86BBD9ac8B9B44C95FFc6BAAe58E25033B7548AA",
+  CYKRW: "0x3c9f5385c288cE438Ed55620938A4B967c080101",
 };
-async function tvl(timestmpa, block) {
-    return {};
+const ibTokens = {
+  IBEUR: "0x96e61422b6a9ba0e068b6c5add4ffabc6a4aae27",
+  IBJPY: "0x5555f75e3d5278082200Fb451D1b6bA946D8e13b",
+  IBGBP: "0x69681f8fde45345C3870BCD5eaf4A05a60E7D227",
+  IBCHF: "0x1CC481cE2BD2EC7Bf67d1Be64d4878b16078F309",
+  IBAUD: "0xFAFdF0C4c1CB09d430Bf88c75D88BB46DAe09967",
+  IBKRW: "0x95dFDC8161832e4fF7816aC4B6367CE201538253"
 };
-module.exports  = {
-    ethereum:{
-        tvl,
-        staking
+const userInfo = {
+  "inputs": [
+    {
+      "internalType": "uint256",
+      "name": "",
+      "type": "uint256"
+    },
+    {
+      "internalType": "address",
+      "name": "",
+      "type": "address"
+    }
+  ],
+  "name": "userInfo",
+  "outputs": [
+    {
+      "internalType": "uint256",
+      "name": "amount",
+      "type": "uint256"
     },
+    {
+      "internalType": "uint256",
+      "name": "rewardDebt",
+      "type": "uint256"
+    }
+  ],
+  "stateMutability": "view",
+  "type": "function"
+};
+
+const totalBorrows = {
+  "constant": true,
+  "inputs": [],
+  "name": "totalBorrows",
+  "outputs": [{
+    "internalType": "uint256",
+    "name": "",
+    "type": "uint256"
+  }],
+  "payable": false,
+  "stateMutability": "view",
+  "type": "function"
+}
+
+async function staking(timestamp, block) {
+
+  const balances = {};
+  // Protocol Credit Mining LPs
+  await sumTokensAndLPsSharedOwners(
+    balances,
+    [[KPR_WETH_SUSHI_POOL, true],
+      [KPR_LDO_SUSHI_POOL, true],
+      [KPR_MM_SUSHI_POOL, true],
+      [KPR_AMOR_SUSHI_POOL, true],
+      [KP3R, false]],
+    [KP3R],
+    block
+  );
+
+  return balances;
+
+}
+
+async function tvl(timestamp, block) {
+  const balances = {};
+
+  const { output: sushiToken } = await sdk.api.abi.call({
+    abi: userInfo,
+    target: MASTERCHEF,
+    params: [58, YEARN_DEPLOYER],
+    block
+  });
+
+  await unwrapUniswapLPs(balances, [{ token: KPR_WETH_SUSHI_POOL, balance: sushiToken.amount }], block);
+  await sumTokensAndLPsSharedOwners(
+    balances,
+    [
+      [MIM, false],
+      [CVX, false],
+      [DAI, false],
+      [KP3R, false],
+      [SEUR, false],
+      [SAUD, false],
+      [SGBP, false],
+      [SKRW, false],
+      [SJPY, false],
+      [SCHF, false]
+    ].concat([
+      [KPR_WETH_SUSHI_POOL, true],
+      [USDC_ibAUD_POOL, true],
+      [USDC_ibEUR_POOL, true],
+      [USDC_ibKRW_POOL, true],
+      [USDC_ibJPY_POOl, true],
+      [USDC_ibGBP_POOL, true],
+      [USDC_ibCHF_POOL, true]
+    ], Object.values(ibTokens).map(t => [t, false])),
```

Comment:
imo might be better to count the collateral backing fixed forex tokens, since that would lead to a methodology that's closer to other similar protocols that mint synthetic assets (eg: for maker and synthetix we count coins backing synths/dai)